src/modules/adaptors/Staking/EtherFiStakingAdaptor.sol

```diff
@@ -0,0 +1,129 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.21;
+
+import { ERC20, SafeTransferLib, Cellar, PriceRouter, Registry, Math } from "src/modules/adaptors/BaseAdaptor.sol";
+import { Address } from "@openzeppelin/contracts/utils/Address.sol";
+import { StakingAdaptor, IWETH9 } from "./StakingAdaptor.sol";
+import { ILiquidityPool, IWithdrawRequestNft, IWEETH } from "src/interfaces/external/IStaking.sol";
+
+/**
+ * @title EtherFi Staking Adaptor
+ * @notice Allows Cellars to stake with EtherFi.
+ * @author crispymangoes
+ */
+contract EtherFiStakingAdaptor is StakingAdaptor {
+    using SafeTransferLib for ERC20;
+    using Math for uint256;
+    using Address for address;
+
+    /**
+     * @notice The EtherFi liquidity pool staking calls are made to.
+     */
+    ILiquidityPool public immutable liquidityPool;
+
+    /**
+     * @notice The EtherFi withdraw request NFT withdraw requests are made to.
+     */
+    IWithdrawRequestNft public immutable withdrawRequestNft;
+
+    /**
+     * @notice The wrapper contract for eETH.
+     */
+    IWEETH public immutable weETH;
+
+    /**
+     * @notice The eETH contract.
+     */
+    ERC20 public immutable eETH;
+
+    constructor(
+        address _wrappedNative,
+        uint8 _maxRequests,
+        address _liquidityPool,
+        address _withdrawRequestNft,
+        address _weETH,
+        address _eETH
+    ) StakingAdaptor(_wrappedNative, _maxRequests) {
+        liquidityPool = ILiquidityPool(_liquidityPool);
+        withdrawRequestNft = IWithdrawRequestNft(_withdrawRequestNft);
+        weETH = IWEETH(_weETH);
+        eETH = ERC20(_eETH);
+    }
+
+    //============================================ Global Functions ===========================================
+    /**
+     * @dev Identifier unique to this adaptor for a shared registry.
+     * Normally the identifier would just be the address of this contract, but this
+     * Identifier is needed during Cellar Delegate Call Operations, so getting the address
+     * of the adaptor is more difficult.
+     */
+    function identifier() public pure virtual override returns (bytes32) {
+        return keccak256(abi.encode("EtherFi Staking Adaptor V 0.0"));
+    }
+
+    //============================================ Override Functions ===========================================
+
+    /**
+     * @notice Stakes into EtherFi using native asset.
+     */
+    function _mint(uint256 amount) internal override {
+        liquidityPool.deposit{ value: amount }();
```

Comment:
What do you think about implementing a `_maxAvailable()` but for native tokens? It would require ensuring some type of buffer amount for gas costs, and use something like `address(this).balance` I guess during `delegateCalls` to strategist functions like this. Not sure if strategists would want this.

Reply:
The max available logic is already in the normal `mint`.

Reply:
Hmm I think I'm referring to using native tokens. Not `primitives`

Reply:
We discussed this, and your decision to leave it due to it being cleaner to always wrap/unwarp the native asset makes sense. Especially as the strategists can handle the few scenarios where they have to deal with the native asset as is.